ambry-router/src/test/java/com.github.ambry.router/GetBlobOperationTest.java

```diff
@@ -0,0 +1,596 @@
+    blobProperties = new BlobProperties(blobSize, "serviceId", "memberId", "contentType", false, Utils.Infinite_Time);
+    userMetadata = new byte[BLOB_USER_METADATA_SIZE];
+    random.nextBytes(userMetadata);
+    putContent = new byte[blobSize];
+    random.nextBytes(putContent);
+    ReadableStreamChannel putChannel = new ByteBufferReadableStreamChannel(ByteBuffer.wrap(putContent));
+    blobIdStr = router.putBlob(blobProperties, userMetadata, putChannel).get();
+    router.close();
+    networkClient = networkClientFactory.getNetworkClient();
+    resetConfigsForGet();
+  }
+
+  /**
+   * Recreate the configs for the router to be used by gets.
+   */
+  private void resetConfigsForGet() {
+    requestParallelism = 2;
+    successTarget = 1;
+    VerifiableProperties vprops = new VerifiableProperties(getNonBlockingRouterProperties());
+    routerConfig = new RouterConfig(vprops);
+  }
+
+  /**
+   * Test {@link GetBlobOperation} instantiation and validate the get methods.
+   * @throws Exception
+   */
+  @Test
+  public void testInstantiation()
+      throws Exception {
+    maxChunkSize = random.nextInt(1024 * 1024) + 1;
+    blobSize = maxChunkSize * 5;
+    doPut();
+    String blobIdStr = (new BlobId(mockClusterMap.getWritablePartitionIds().get(0))).getID();
+    Callback<ReadableStreamChannel> operationCallback = new Callback<ReadableStreamChannel>() {
+      @Override
+      public void onCompletion(ReadableStreamChannel result, Exception exception) {
+        // no op.
+      }
+    };
+
+    // test a bad case
+    try {
+      new GetBlobOperation(routerConfig, routerMetrics, mockClusterMap, responseHandler, "invalid_id", operationFuture,
+          operationCallback, operationCompleteCallback, time);
+      Assert.fail("Instantiation of GetReadableStreamChannel operation with an invalid blob id must fail");
+    } catch (RouterException e) {
+      Assert.assertEquals("Unexpected exception received on creating GetBlobOperation", RouterErrorCode.InvalidBlobId,
+          e.getErrorCode());
+    }
+
+    // test a good case
+    GetBlobOperation op =
+        new GetBlobOperation(routerConfig, routerMetrics, mockClusterMap, responseHandler, blobIdStr, operationFuture,
+            operationCallback, operationCompleteCallback, time);
+
+    Assert.assertEquals("Callbacks must match", operationCallback, op.getCallback());
+    Assert.assertEquals("Futures must match", operationFuture, op.getFuture());
+    Assert.assertEquals("Blob ids must match", blobIdStr, op.getBlobIdStr());
+  }
+
+  /**
+   * Put blobs that result in a single chunk; perform gets of the blob and ensure success.
+   */
+  @Test
+  public void testSimpleBlobGetSuccess()
+      throws Exception {
+    maxChunkSize = random.nextInt(1024 * 1024) + 1;
+    for (int i = 0; i < 10; i++) {
+      // blobSize in the range [1, maxChunkSize]
+      blobSize = random.nextInt(maxChunkSize) + 1;
+      doPut();
```

Comment:
once again, personal pref - but i would find the tests *much* more readable if `doPut()` just took the arguments it needed instead of us setting the arguments via class variables. 

One of the reasons I say that is because if I want to reuse the function and want to know what its knobs are, I *have* to read the whole function. If it took arguments, I would simply read the description of the arguments and know what I need.

Any new arguments will also need to be added as class variables and that might make it more confusing.